recipe/bld.bat

```diff
@@ -5,11 +5,12 @@ set ARROW_BUILD_TOOLCHAIN=%LIBRARY_PREFIX%
 
 cmake -G "%CMAKE_GENERATOR%" ^
       -DCMAKE_INSTALL_PREFIX="%LIBRARY_PREFIX%" ^
-      -DARROW_BOOST_USE_SHARED:BOOL=OFF ^
+      -DARROW_BOOST_USE_SHARED:BOOL=ON ^
       -DARROW_BUILD_TESTS:BOOL=OFF ^
       -DARROW_BUILD_UTILITIES:BOOL=OFF ^
-      -DCMAKE_BUILD_TYPE=Release ^
-      -DARROW_PYTHON=on ^
+      -DCMAKE_BUILD_TYPE=release ^
+      -DARROW_PYTHON:BOOL=ON ^
+      -DARROW_ORC:BOOL=ON ^
```

Comment:
FYI this is automatically turned off at build script generation time, but presumably we'd like to get this working at some point.